### ISSUE TITLE
Long lines are sometimes not reported as E501 error

### ISSUE THREAD

**Issue by tlinhart** [astral-sh/ruff#5248]
Consider this code in `test.py`:

```
def f(argument_with_a_very_long_name):
    pass


aaaaaaaaaa_bbbbbbbbbb_cccccccccc_dddddddddd_eeeeeeeeee_ffffffffff = "xxx"

f(
    argument_with_a_very_long_name=aaaaaaaaaa_bbbbbbbbbb_cccccccccc_dddddddddd_eeeeeeeeee_ffffffffff
)
```

and these Ruff settings in `pyproject.toml`:

```
[tool.ruff]
cache-dir = "~/.cache/ruff"
select = ["E", "F", "W", "I"]
line-length = 79
target-version = "py310"
```

When I run Ruff with `ruff check test.py`, it doesn't report any violation despite the fact that the code contains a line exceeding the length of 79 characters (line 8). On the other hand, running `flake8 test.py` reports:

```
test.py:8:80: E501 line too long (100 > 79 characters)
```

Regarding the line too long violations, I only encoutered [this FAQ paragraph](https://beta.ruff.rs/docs/faq/#is-ruff-compatible-with-black) about compatibility with Black where it states:

> Ruff, on the other hand, will flag rule E501 for any line that exceeds the line-length setting.

From that I would assume that Ruff should report the issue in all cases.

I use Ruff version 0.0.274.

**Comment by charliermarsh** [comment]
This is intended -- we don't flag lines that consistent of a single unbroken word. I can add a note to the documentation.

**Comment by tlinhart** [comment]
Thanks for clarifying. On one hand, it makes sense as it makes it consistent with Black's best effort behavior. On the other hand, coming from Flake8, it makes a lot of `# noqa: E501` comments obsolete.
